### ISSUE TITLE
UnicodeEncodeError in furl.join using Python 2.7 

### ISSUE THREAD

**Issue by jeffbr13** [gruns/furl#58]
Loving furl, it's a great interface for URL-handling! Found this inconsistency in the API while using Python 2.7:

e.g.

```python
# works fine
>>> furl(u'http://example.org/?kødpålæg=42')
furl('http://example.org/?k%C3%B8dp%C3%A5l%C3%A6g=42')

# fails
>>> furl('http://example.org').join(u'/?kødpålæg=42')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    furl('http://example.org').join(u'/kødpålæg=42')
  File "/Users/jeffbr/.virtualenvs/clipper/lib/python2.7/site-packages/furl/furl.py", line 1260, in join
    self.load(urljoin(self.url, str(url)))
UnicodeEncodeError: 'ascii' codec can't encode character u'\xf8' in position 2: ordinal not in range(128)
```

Related to <https://github.com/gruns/furl/issues/33> where you fixed `furl.__init__`, but no sanitation seems to be done for Python 2's unicode strings in [`urljoin`](https://github.com/gruns/furl/blob/7ea8d6a8a98cf5719d49d95cf6874f0ca82fe6bd/furl/furl.py#L1344)?

**Comment by gruns** [comment]
Fixed in furl v0.4.91. Update with

```
$ pip install furl --upgrade
```

Thank you for bringing this issue to my attention Ben. Don't hesitate to let me know if there's anything else I can do.

**Comment by jeffbr13** [comment]
Brilliant, cheers!
